### PR TITLE
Added type definitions for amap-js-api-indoor-map

### PR DESCRIPTION
**Pull request by breeze9527**
… amap-js-api)

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/amap-js-api-indoor-map/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+            id: string;
+            status: 1;
+            error: Error;
+        }
+        type SearchResult = SearchSuccessResult | SearchErrorResult;
+    }
+
+    class IndoorMap extends Layer {
+        constructor(options?: IndoorMap.Options);
+        showIndoorMap(
```
Comment (sheetalkamat) [inline types/amap-js-api-indoor-map/index.d.ts]: You want your overloads in reverse order with fewer parameters coming first.

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

File: types/amap-js-api-indoor-map/index.d.ts (hunk 1 above)
Comment (breeze9527) [reply types/amap-js-api-indoor-map/index.d.ts]: Thank for your review, and class `Layer` is a abstract class and it has no constructor function, should i write a empty constructor function in class `Layer` ?

Comment (sheetalkamat) [reply types/amap-js-api-indoor-map/index.d.ts]: I meant order of overloads for showIndoorMap not the constructor

Comment (breeze9527) [reply types/amap-js-api-indoor-map/index.d.ts]: Updated 😄

**Review by sheetalkamat — APPROVED** [review]

**Pull request merged**